test/test_macro.hpp

```diff
@@ -0,0 +1,58 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2022 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#pragma once
+
+#include "test.hpp"
+
+#define CHECK(...)                                    \
```

Comment:
Why this is copied from test.hpp?

Reply:
> Why this is copied from test.hpp?

@atamazov  this idea is to separate FAIL() out in a separate header which conflicts with the same definition in GoogleTest. It seems to be strange if just move FAIL() out, that's I put all Macros in the new header file

Reply:
@xinlipn But then why test.hpp (that contains the same CHECK) is included here?

Reply:
> @xinlipn But then why test.hpp (that contains the same CHECK) is included here?

@atamazov separate out only one Macro FAIL() from the rest and just create a header file for it seems to be a temp workaround and looks weird logically, at least to me. 

FAIL() isn't used anywhere and I removed it for now to minimize the impact to other code.
Consequently test_macro.hpp is removed and changes in other headers anyway are reverted.